File: src/dataset/cpp/meetup_exercism-toemm/claude/ClaudeSonnet4_meetup_v3.cpp
#include "meetup.h"

#include <boost/date_time/gregorian/gregorian.hpp>

namespace meetup {

using namespace boost::gregorian;

scheduler::scheduler(months_of_year month, int year) : 
    _month(month), 
    _year(year), 
    teenths(date_period(date(_year, _month, 13), days(7))) {
    precompute_teenth_dates();
}

void scheduler::precompute_teenth_dates() {
    for (auto it = teenths.begin(); it != teenths.end(); it += days(1)) {
        teenth_dates[it.day_of_week().as_number()] = it.day();
    }
}

date scheduler::get_teenth_by_weekday(greg_weekday weekday) const {
    return date(_year, _month, teenth_dates[weekday.as_number()]);
}

date scheduler::monteenth() const {
    return get_teenth_by_weekday(Monday);
}

date scheduler::tuesteenth() const {
    return get_teenth_by_weekday(Tuesday);
}

date scheduler::wednesteenth() const {
    return get_teenth_by_weekday(Wednesday);
}

date scheduler::thursteenth() const {
    return get_teenth_by_weekday(Thursday);
}

date scheduler::friteenth() const {
    return get_teenth_by_weekday(Friday);
}

date scheduler::saturteenth() const {
    return get_teenth_by_weekday(Saturday);
}

date scheduler::sunteenth() const {
    return get_teenth_by_weekday(Sunday);
}

}  // namespace meetup

File: src/dataset/python/meetup_exercism-fruit-in/claude/ClaudeSonnet4_meetup_v3.py
from calendar import Calendar
from datetime import date

def meetup(year, month, week, day_of_week):
    target_weekday = ['Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday', 'Sunday'].index(day_of_week)
    
    if week == 'teenth':
        for day_num, weekday in Calendar().itermonthdays2(year, month):
            if day_num != 0 and weekday == target_weekday and 13 <= day_num <= 19:
                return date(year, month, day_num)
    elif week == 'last':
        last_day = None
        for day_num, weekday in Calendar().itermonthdays2(year, month):
            if day_num != 0 and weekday == target_weekday:
                last_day = day_num
        if last_day is not None:
            return date(year, month, last_day)
    else:
        occurrence = int(week[0])
        count = 0
        for day_num, weekday in Calendar().itermonthdays2(year, month):
            if day_num != 0 and weekday == target_weekday:
                count += 1
                if count == occurrence:
                    return date(year, month, day_num)
        raise MeetupDayException(r".+")

class MeetupDayException(Exception):
    pass